src/config/gen.ts

```diff
@@ -1,227 +1,30 @@
-      }
-      // const specific = await genPartialConfigFromDryInstances(instancesDry, args)
-      const common = _.merge(
-        {},
-        defaultCommonConfig,
-        await T.collapseFort(expand(commonFort), args)
-      )
-
-      return _.merge(
-        {},
-        specific,
-        common,
-      )
-    }
-  }
-
-const validateInstancesType = (instances: T.EitherInstancesConfig, msg: string = '') => {
-  if (_.isArray(instances)) {
-    T.decodeOrThrow(T.DryInstancesConfigV, instances, 'Could not validate Instances Array')
-  } else if (_.isObject(instances)) {
-    T.decodeOrThrow(T.SugaredInstancesConfigV, instances, 'Could not validate Instances Object')
-  }
-}
-
-
-/**
- * 1. If a dna config object contains a URL in the path, download the file to a temp directory,
- *     and rewrite the path to point to downloaded file.
- * 2. Then, if the hash is not set, calculate the hash and set it.
- * 3. Add the UUID for this scenario
- */
-export const resolveDna = async (inputDna: T.DnaConfig, providedUuid: string): Promise<T.DnaConfig> => {
-  const dna = _.cloneDeep(inputDna)
-
-  dna.id = dna.uuid ? `${dna.id}::${dna.uuid}` : dna.id
-  dna.uuid = dna.uuid ? `${dna.uuid}::${providedUuid}` : providedUuid
-
-  if (!dna.hash) {
-    dna.hash = !env.legacy ? "" : await getDnaHash(dna.file, dna.uuid).catch(err => {
-      logger.warn(`Could not determine hash of DNA at '${dna.file}'. Note that tryorama cannot determine the hash of DNAs at URLs\n\tOriginal error: ${err}`)
-      return "[UNKNOWN]"
-    })
-  }
-  return dna
-}
-
-export const getConfigPath = configDir => path.join(configDir, 'conductor-config.yaml')
-
-export const desugarInstances = (instances: T.SugaredInstancesConfig, args: T.ConfigSeedArgs): T.DryInstancesConfig => {
-  T.decodeOrThrow(T.SugaredInstancesConfigV, instances)
-  // time to desugar the object
-  return Object.entries(instances).map(([id, dna]) => ({
-    id,
-    agent: makeTestAgent(id, args),
-    dna
-  } as T.DryInstanceConfig))
+  // what goes in here?
 }
 
-export const makeTestAgent = (id, { playerName, uuid }) => ({
-  // NB: very important that agents have different names on different conductors!!
-  name: `${playerName}::${id}::${uuid}`,
-  id: id,
-  keystore_file: '[UNUSED]',
-  public_address: '[SHOULD BE REWRITTEN]',
-  test_agent: true,
-})
-
-/*
-TODO: useful for dry happBudles?
-export const genPartialConfigFromDryInstances = async (instances: T.DryInstancesConfig, args: T.ConfigSeedArgs) => {
-
-  // Here we:
-  // - generate (much smaller) config
-  // -
-
+export const gen = (): T.ConfigSeed => (
+  args: T.ConfigSeedArgs
+): T.RawConductorConfig => {
   const { configDir, adminInterfacePort, uuid } = args
 
-  const config: any = {
-    agents: [],
-    dnas: [],
-    instances: [],
+  const specific: any = {
     environment_path: configDir,
+    admin_interfaces: [
+      {
+        driver: {
+          type: 'websocket',
+          port: 8000,
```

Comment:
this being hardcoded is a TODO... 
relates to the fact that I don't know how to set each adminInterfacePort for different conductors without conflicts

Reply:
For the future, we can set it to `0` and then provide a way to discover the port that was selected (probably via stdin).

Or, use the `get-port-cautiously` module to get a free port

Reply:
oh gotcha, I just discovered we already had that module in use, but I didn't see it before, latest code changes make use of get-port-cautiously